src/tests/ftest/cart/test_group_np_common.h

```diff
@@ -187,6 +187,8 @@ test_swim_status_handler(crt_rpc_t *rpc_req)
 		  e_req->rank, swim_seq,
 		  e_req->exp_status);
 
+	D_FREE(swim_seq);
```

Comment:
Let's avoid a coverity bug before it happens.  You have never tested that the malloc was successful so you don't know if you can free it.  The code needs to check swim_seq at the top of this function.

Reply:
I was going to say, we check that in D_ALLOC but we didn't use D_ALLOC...so there is another issue here.  We should not mix and match malloc and D_FREE or D_ALLOC and free.  Be consistent.

However, malloc will return NULL if it fails and free() can be called with a NULL pointer so there is no issue there.   But if we don't check, there probably is an issue elsewhere.

Reply:
You may be able to call free with the pointer being NULL, but you also use the pointer to call memset and strcpy.  Do not depend on all usage of this pointer to accept NULL.

Reply:
yeah, I agree we need to check the return of malloc.  I was mostly pointing out the mismatch.

Reply:
> yeah, I agree we need to check the return of malloc. I was mostly pointing out the mismatch.

How about this?

```diff
+++ b/src/tests/ftest/cart/test_group_np_common.h
@@ -148,6 +148,10 @@ test_swim_status_handler(crt_rpc_t *rpc_req)
        static const char               *alive_regex = ".?0*";
        char                            *swim_seq = malloc(MAX_SWIM_STATUSES);

+       D_ASSERTF(swim_seq != NULL,
+                 "Cannot allocate %d chars for SWIM sequence.\n",
+                 MAX_SWIM_STATUSES);
+
        /* CaRT internally already allocated the input/output buffer */
        e_req = crt_req_get(rpc_req);
```

Reply:
Since you know the size to malloc at compile time and since the alloc/free is self-contained to just this function, then why don't you declare the swim_seq just as a local variable of swim_seq[MAX_SWIM_STATUS] and forget about the alloc and free.  This will make the code cleaner and save time.